iDiff/utils/image_prep_utils.go

```diff
@@ -129,32 +89,47 @@ type CloudPrepper struct {
 }
 
 func (p CloudPrepper) ImageToFS() (string, error) {
-	// check client compatibility with Docker API
-	valid, err := ValidDockerVersion()
+	URLPattern := regexp.MustCompile("^.+/(.+(:.+){0,1})$")
+	URLMatch := URLPattern.FindStringSubmatch(p.Source)
+	path := strings.Replace(URLMatch[1], ":", "", -1)
+	ref, err := docker.ParseReference("//" + p.Source)
+	if err != nil {
+		panic(err)
+	}
+
+	img, err := ref.NewImage(nil)
 	if err != nil {
+		glog.Error(err)
 		return "", err
 	}
-	var tarPath string
-	if !valid {
-		glog.Info("Docker version incompatible with api, shelling out to local Docker client.")
-		imageID, imageName, err := pullImageCmd(p.Source)
+	defer img.Close()
+
+	imgSrc, err := ref.NewImageSource(nil, nil)
+	if err != nil {
+		glog.Error(err)
+		return "", err
+	}
+
+	if _, ok := os.Stat(path); ok != nil {
+		os.MkdirAll(path, 0777)
+	}
+
+	for _, b := range img.LayerInfos() {
+		bi, _, err := imgSrc.GetBlob(b)
 		if err != nil {
-			return "", err
+			glog.Error(err)
```

Comment:
I expanded the error messages to hopefully be a bit more descriptive.  We definitely could stop if there is a layer unable to be pulled, but there could be some benefit potentially of seeing which layers fail in case they are malformed? If they can be malformed?